lesson18/index.js

```diff
@@ -0,0 +1,186 @@
+        Button.classList.add("pagination__btn","js-pagination-btn");
+        Button.dataset.index = `${i}`;
+
+        //JSONデータでdisplay:trueの場合はis-activeを付与
+        data[i].display && li.classList.add("is-active");
+
+        fragment.appendChild(li).appendChild(Button);
+    }
+    pagination.insertAdjacentElement("afterbegin",paginationList).appendChild(fragment);
+}
+
+function init(data) {
+    const allofCountElement = document.getElementById("js-counter-all");
+    
+    allofCountElement.textContent = data.length;
+    incrementCurrentIndex(getCurrentIndex());
+    toggleButtonDisabled(data.length);
+}
+
+function toggleButtonDisabled(length) {
+    const currentIndex = getCurrentIndex();
+    const firstIndex = 0;
+    const lastIndex = length - 1;
+
+    previousButton.disabled = currentIndex === firstIndex;
+    nextButton.disabled = currentIndex === lastIndex;
+}
+
+function getCurrentIndex() {
+    const activeImg = ul.querySelector(".is-active");
+    return Number(activeImg.dataset.index);
+}
+
+function incrementCurrentIndex(number) {
+    const currentCountElement = document.getElementById("js-counter-current");
+    currentCountElement.textContent = number + 1;
+}
+
+function switchImg(number) {
+    const activeImg = ul.querySelector(".is-active");
+    const imgItems = [...document.getElementsByClassName("js-slideshow-item")];
+    activeImg.classList.remove("is-active");
+    imgItems[number].classList.add("is-active");
+}
+
+function switchPagination(number) {
+    const ul = document.getElementById("js-pagination-list");
+    const paginationItems = [...document.getElementsByClassName("js-pagination-item")];
+    const activeItem = ul.querySelector(".is-active");
+    activeItem.classList.remove("is-active");
+    paginationItems[number].classList.add("is-active");
+}
+
+const addEventListenerForNextButton = (length) => {
+    nextButton.addEventListener ("click", () => {
+        switchImg(getCurrentIndex() + 1);
+        switchPagination(getCurrentIndex());
+        incrementCurrentIndex(getCurrentIndex());
+        toggleButtonDisabled(length);
+    })
+}
+
+const addEventListenerForPreviousButton = (length) => {
+    previousButton.addEventListener ("click", () => {
+        switchImg(getCurrentIndex() - 1);
+        switchPagination(getCurrentIndex());
+        incrementCurrentIndex(getCurrentIndex());
+        toggleButtonDisabled(length);
+    })
+}
+
+const addEventListenerForPagination = (length) => {
+    const paginationList = document.getElementById("js-pagination-list");
+
+    paginationList.addEventListener ("click", (e) => {
+
+        //buttonとbuttonの間はクリック対象外にする
+        if (paginationList && e.currentTarget !== e.target) {
+            const clickedItemIndex = Number(e.target.dataset.index);
+
+            switchImg(clickedItemIndex);
+            switchPagination(clickedItemIndex);
+            incrementCurrentIndex(clickedItemIndex);
+            toggleButtonDisabled(length);
+        }
+    })
```

Comment:
IMO
If I write the condition as commented out, it would look like this, what do you think?
```suggestion
  paginationList.addEventListener("click", (e) => {
    //buttonとbuttonの間はクリック対象外にする
    if (e.currentTarget === e.target) {
      return;
    }
    const clickedItemIndex = Number(e.target.dataset.index);
    switchPagination(clickedItemIndex);
    incrementCurrentIndex(clickedItemIndex);
    switchImg(clickedItemIndex);
    toggleButtonDisabled(length);
  });
```

Reply:
Wow! 
That was an eye opener!
It made the code very easy to understand. Thank you!
[e573c18](https://github.com/haru-programming/JavaScript-lessons/pull/22/commits/e573c18852554f6914b2a73af4512c85c751e05e)